### DISCUSSION TITLE
12v power via EcoNet

### DISCUSSION THREAD

**Discussion by rexbron** [esphome-econet/esphome-econet#118] [Q&A] [answered]
The instructions mention wiring up 3 of the 4 conductors. Is there 12v available over the RJ11 connector to power the ESP board with a suitable voltage converter?

**Comment by barndawgie** [comment]
We haven't been able to figure that out (and some folks have tried). If you are able to figure out a way to power the device via the RJ11 connector you will make a lot of folks here happy and we can update our guides.

**Comment by tronikos** [comment, marked as answer]
Unfortunately not. But if you are comfortable accessing the control board you can do what elmoret did at
https://discord.com/channels/1148015790038188073/1148327349620838511/1162186716635988009
I copied their discord messages and photos at our wiki https://github.com/esphome-econet/esphome-econet/wiki/Powering-ESP-from-the-water-heater-control-board

**Comment by fireheadman** [comment]
this post needs better pictures... which I can help with.
I have everything disassembled and ready to connect, but there is nothing that explains the connection.

The picture does not provide a good angle showing if it is 3 wires or 4 wires connected.

If anyone can give some info to this, I'll snap some better pictures

**Comment by fireheadman** [comment]
Think I might have found the reference I needed....
I'm using the same M5 Stack module, which comes with a orange 4-pin connector.


A typical phone cord should be this color (prong/clip facing up)
Left to Right   -->  RS-485 orange slot
-------------
```
Pin 1 - Black  --> (G)
Pin 2 - Red    --> (A)
Pin 3 - Green   --> (B)
Pin 4 - Yellow  --> Cut and solder to board 12v test pin
```      

**This picture shows this...**
![image](https://github.com/esphome-econet/esphome-econet/assets/19847195/03c18a62-21ba-4d17-815f-f9fb031a38ce)

Here is my setup, I'm using an old Monster Cable phone cable I found in my cord stash.
This cord is a pretty nice upgrade from a standard cord, it has stranded wire instead of solid core, which will handle voltage a little cleaner.   I will wrap this in some cloth tape and heat shrink the ends to clean it up (picture coming later on this)
**_EDIT:   Ignore the wire connection to the RS485 in this picture, I had the black and green flipped_**
![image](https://github.com/esphome-econet/esphome-econet/assets/19847195/0dcb8acf-f914-4585-be53-d80a547adf1f)

**Here is a close up on the pin out...**
What I did was cut the cord/wires twice the length needed so I could clip the yellow and route it back to the 12v test pin on the board.
![image](https://github.com/esphome-econet/esphome-econet/assets/19847195/3845e3f8-ffaf-4d33-96e8-c6d65a8e0c34)

**Close up on the board to where to solder this wire**
![image](https://github.com/esphome-econet/esphome-econet/assets/19847195/fa9b271e-1394-4d4c-8ba2-08b980a5f6f2)

**Comment by fireheadman** [comment]
Here's the end result...   Nice and clean setup
2 things that scared the Shiet out of me....

1)  I removed the board from the plastic, which involves gently disconnecting the 2 ribbon cables after releasing the cable stay/clip.
When I reinstalled them, I didn't have the larger one all the way seated, so half of the touch controls weren't working.   Instant knot in my stomach on that one!   Reseated it and eveything is good (just a few more gray hairs).

2)  When I programmed the M5, I used the web installer (https://esphome-econet.github.io/install/)  which took all over 3 minutes to complete.   Very nice!!!   During the programming the M5 had a green light on it.    After the programming, I disconnected it.    Plugging it back in I noticed the light no longer comes on, so I though maybe it was fried somehow?   But it connects to home assistant fine.   I even re-flashed it with no issues.   

**Curious why the light is programmed to stay off and how to get it back on?**

Used a small command strip to mount it.
![image](https://github.com/esphome-econet/esphome-econet/assets/19847195/f8ae39c8-35cb-4def-b255-760dbc3efb5d)

**Comment by fireheadman** [comment]
some info from the integration

![image](https://github.com/esphome-econet/esphome-econet/assets/19847195/dc951590-a46d-4848-a4be-b470540f61b2)
![image](https://github.com/esphome-econet/esphome-econet/assets/19847195/716c34ab-0186-4512-acd1-d31bb8d37ea4)

**Comment by eigenphase** [comment]
This is awesome, thank you! I’ll try this.

@fireheadman , Questions:

1. It seems you fiddle the wire out. Couldn’t you just solder it to the RJ connector on the board? Then Pin4 would always have supply. Only RJ11  cable needed
2. any tip on how I remove the black controller box while minimizing damage?
3. Your Hass has so many nice options!! Would you mind sharing your yaml?

edit: Re 3, seems you used the new install wizard. Wow, so much changed!! I still have a very early config

**Comment by fireheadman** [comment]
guess you could, but if I ever need to revert the modification, i just unsolder it.
If I cut a port connector, then I am SOL.

as for the trim bezel, just be gentle with it and it will pop off.... I used a plastic pry tool from my car audio tool set.   Could also use a plastic putty blade.

...and for sharing yaml, I am setup with a split configuration, so there is over 50 yaml files....  its been an evolution over the past 4 years.  This might be a little overwhelming to share.

and yeah, the web installer is a breeze!   It even imported without any issues into ESPHome Addon.

**Reply by eigenphase** [reply]
> guess you could, but if I ever need to revert the modification, i just unsolder it.
If I cut a port connector, then I am SOL.

what I mean is to solder the other end to the existing RJ12 connector. It’s exactly the same modification (=soldering a wire to an existing pad) as you’re doing anyway. Just on two sides except one.

> as for the trim bezel, just be gentle with it and it will pop off.... I used a plastic pry tool from my car audio tool set. Could also use a plastic putty blade.

I really have no clue. I guess first flip the breaker. No screws or anything. Between tank and the entire plastic enclosure put a thin blade? Or only the TOP (outermost part) of the plastic thing?

And where exactly pry ? Sides? Top, bottom?
I tried gently pulling but it didn’t move for me. I really want to avoid breaking some plastic :-(

**Comment by fireheadman** [comment]
Start on the right side, this is where the clips are (believe 2 of them).  Gently wedge whatever tool you want to pry with just a little bit under it and it should pop off.

The left side does not clip, it has prongs that is slides into...
The only thing this cover does is shield the back battery (little coin battery) and access to the 2 screws.   

And back to the soldering to the RJ11 port, unless you clip the wire you intend to solder to (which is a one way street)... then you still have to know where on the board that leg goes to.    You wouldn't want to energize a circuit with 12v and have a blue smoke generator on your hands.   I played it safe.

**Comment by fireheadman** [comment]
Here's some additional tweaks you can do...
added this to the esphome yaml for the water heater

```
light:
  - platform: fastled_clockless
    chipset: WS2812B
    pin: 27
    num_leds: 1
    rgb_order: GRB
    id: status_led
    name: ${friendly_name} Light
    effects:
      - random:
      - flicker:
      - addressable_rainbow:
      - pulse:
          name: "Slow Pulse"
          update_interval: 2s
      - pulse:
          name: "Fast Pulse"
          transition_length: 0.5s
          update_interval: 0.5s
          min_brightness: 0%
          max_brightness: 100%

binary_sensor:
  - platform: gpio
    pin:
      number: 39
      inverted: true
    name: ${friendly_name} Button
    on_press:
      then:
        - light.toggle: status_led
```

...added those and pushed to the M5 Stack, now I have a controllable LED to help me with alerts and automations.
Maybe keep it on a slow pulse (heartbeat) and blue when healthy
Then change to red when/if an alert is raised or orange when hot water availability is below 10%.

**Reply by BigThunderSR** [reply]
Looks like this is the solution for https://github.com/esphome-econet/esphome-econet/issues/241

Thanks!

**Reply by fireheadman** [reply]
Glad to help

**Reply by k8gg** [reply]
Fast forward to 2026. Today I tried to compile and `fastled_clockless` does not work any more, and I would have to switch to `esp32_rmt_led_strip`. 
With `fastled_clockless` it compiled OK back in early Jan. 2026.

Now, is this because we are now using `esp-idf` and no longer `arduino` as the framework these days...? 
Now, I see there was a PR #415 saying we are moving to esp-idf, but then these yaml files under build-yaml [here](https://github.com/esphome-econet/esphome-econet/tree/main/build-yaml) still reads arduino... so I guess I'm confused.

**Comment by eigenphase** [comment]
@fireheadman I got the control unit off but there is this nasty plastic crap in the back. It just doesn’t move and seems to be fixed at the two screw holes.

how do I remove it?

![IMG_3055](https://github.com/esphome-econet/esphome-econet/assets/7464149/5af7992e-e8d4-4376-a89e-a656e0785dcf)

**Reply by CTho9305** [reply]
Wiggling it while pulling helps a lot.

**Comment by eigenphase** [comment]
I somehow made it work to solder the wire without taking off the plastic crap.

But now I’ve got a bigger issue: The ESP is not coming up.

I measure 13VDC on the wire.

if I remove the Atom from the RS485 board, I measure ~1V at the 3V3 pad and ~20mV on the 5V pad.

is there anything else needed?

**Comment by fireheadman** [comment]
That piece just pops off.   I used my pry tool to pop it loose and then it slowly wiggles off....  its a friction fit so be gentle with it.
If you soldered to the 12v test pad and you have voltage, you should be able to boot it right now.     Can always remove the atom lite from the rs485 base and see if it powers up using a usbc cable.   Verify it boots that way first.... can even add it to HA too, it will just show no values on the ESPHome integration..

Also verify the wires are in the correct locations on the orange header... I messed this up the first time, which I mentioned in the pictures I posted.

**Reply by eigenphase** [reply]
Wow this piece was really stuck as if it would be glued for me :( I guess you pried it next to the two screw holes?

Anyway, if I’m not doing something terribly wrong my Atom RS485 is broken! Very hard to believe but if I power it via USB everything works.

I removed the ESP and the cover and it seems to be that there is just a aoz1282ci that coverts 12V to 5V. If I understand correctly, that 5V is fed to the Atom Mini Board, which generates the 3.3V and the 3.3V are then used for the transceiver.

On the VIN pin I measure 13V, as expected. But on LX (and out) just around 20mV.

If something is broken, I would expect the voltage to drop … but it’s nice and clear +13V (the regulator supports up to 36V). So it really doesn’t make any sense to me except the regulator is broken … but what are the odds??

If easy for you, would you mind removing the Atom Mini and just measuring the 5V pin on the RS485 board? Just to make sure you’re getting 5V there …

**Comment by fireheadman** [comment]
hmmm... ya lost me there.
but maybe just try applying power with usbc cord (leave the 12v wire from water heater board to the side for now)...
see if it boots fine and gets readings from the 3 wires instead.... that was the original method to hook this thing up and would help determine if everything is working as intended.

I'm not able to measure any voltages on mine for a while with the holidays a few days away (can't risk an accident).

**Comment by Here-Be-Dragons** [comment]
I have an older model - Rheem HB50, and pin 6 in the schematic outputs 12vdc which powers my Atom RS485 board. Also, my water heater has an RJ45 jack despite my prior Econet adapter being RJ11.

```
    6P4C RJ11/RJ12 male connector end view   
    
               +---------+
            1  ---       |
            2  ---       +--+      
        B   3  ---          |     
        A   4  ---          |        
       GND  5  ---       +--+
     12vdc  6  ---       |
               +---------+
```
![Screenshot 2023-12-29 161009](https://github.com/esphome-econet/esphome-econet/assets/16006239/78d3286b-2a01-47de-b9f2-ee75549665b0)
![Screenshot 2023-12-29 160952](https://github.com/esphome-econet/esphome-econet/assets/16006239/dca10755-db13-42f2-8655-635fcba15a63)

**Reply by elreydetodo** [reply]
I can confirm that my water heater also has +12VDC on pin 6. I am now running an M5 Atom + RS485 hat using just the wires in the cable, no need to solder anything. My model is PRO H65 T2 RU350BM.

I didn't check closely to see if the port was actually RJ45, but man that would be more convenient than snipping a phone wire. Those little tiny strands of copper were super annoying to work with. I will check that later, and if it _is_ an RJ45 jack I'm going to make a new wire.

**Comment by zarzycki** [comment]
FWIW, I was able to do this earlier today. Thanks for the instructions -- I would have had to run a pretty long USB cable to get things to work in my basement; this is way cleaner.

I had to swap out the board because a batch of new Rheem HPWH that shipped in ~November was throwing erroneous errors (T009), so Rheem sent me a new one and authorized me to install it myself.

My process was simple.

1. Flip the breaker off.
2. Pop the outer bezel off by digging nails in the groove between the front and the RJ11 jack. I went up and down the right side, and the left came unhooked. There is no need for other tools, but a nylon car tool would work.
3. Unscrew the top and bottom Philips screws.
4. Take numerous pictures of the existing wiring for matching later.
5. Unclip everything. I used needle-nose pliers in a couple of places; just make sure you depress the tabs when needed.
6. Take the board to my soldering station
7. As others have noted, a plastic piece covers the board from the back. It is luckily not glued -- it friction fits over the two screw holes (if you look closely at the underside, you'll see two rings around where the screw goes). It was stuck pretty well, but I had success slowly "rocking" first the top two corner points up and then the bottom two corner points up. Slowly, I could see the plastic slide up the interior ring, and eventually, it popped off.
8. Then, it was just a matter of quick soldering (I used some spare 20AWG) to the 12V pin. I ran the wire out through a little gap next to the RJ12 plug (I think I first popped the other end of the wire out through the jack itself and then worked it down the side so it wouldn't interfere with clipping in the RJ12 cable).
9. Pop the back plastic shroud back on over the two screw holes.
10. Head back to HPWH, use photos, and reattach everything as before.
11. Strip other end of 20AWG and plug into 12V on the M5 Atom.
12. Flip the HPWH breaker on. Profit.

**Comment by daspiderboris** [comment]
Posted this on the discord as well, but I thought it would be worth having it here as this was the first page I found when looking for a solution to power the board:

> For Heat Pump Water Heaters, there are a few ways to power the M5. I don't think I've seen this option, but had success on my unit - XE65T10H45U1 

>The GRID module only used 3 out of 4 pins on the connector. I tested the 4th pin and got 13V and was able to power the M5 by adding a breadboard jumper wire. No soldering, no need to disconnect the board from the water heater (obviously power down the water heater first).

>Now to 3d print a case to clean up the spaghetti
![hpwh_v01](https://github.com/user-attachments/assets/1e6e02d8-51c8-49b7-ba18-dd58194de1ea)
![hpwh_v02](https://github.com/user-attachments/assets/8a0b2770-afef-4913-bf21-de0dbc531542)

**Comment by saiarcot895** [comment]
For a traditional Rheem tank water heater (not a heat-pump model), I don't see a 12V header on the PCB. Am I correct in guessing that there is only a 3.3V header available?

![IMG_20250801_231031_159-stripped](https://github.com/user-attachments/assets/5cecf6b2-be6a-4b3c-902c-31f172d10585)

**Comment by eldridge** [comment]
@saiarcot895 I have the exact same board and I had success with using the first pin of J10.  I used my multimeter to test all of the pins on that header while the unit was operating and found that the first pin always had around ~12-13V on it.  I have no idea what that header is used for, so let this be a warning: I am assuming it only provides power, but that's only an assumption: it's based solely on observation, NOT by reading a schematic.

My unit is a 50 gallon Rheem Gladiator (model # XE50T12CS55U1).  Here are the photos I took during the process showing how to pry the board out and how I wired a DuPont connector to J10 through the housing.  The fit can be a little snug due to all of the insulation, so be careful that you don't short anything out when reattaching the panel to the unit.

![pry-points](https://github.com/user-attachments/assets/ce86cc2c-0866-4090-9e16-11951883654f)
![board-showing-j10](https://github.com/user-attachments/assets/ea809b55-c6ff-408c-a3f8-ae8c537be274)
![dupont-to-j10](https://github.com/user-attachments/assets/ff67ae5f-2312-4aa0-9aca-e09a380a23dd)
![wire-out-port](https://github.com/user-attachments/assets/eb2a8bc9-760b-407a-8674-c2360c4547eb)
![power-pin-plugged-in](https://github.com/user-attachments/assets/48bc9eaf-964c-4a69-ba2c-c04c0b15c812)

**Reply by eldridge** [reply]
Oh, and I forgot to mention -- although the modular connector on this board is 6P6C rather than 4P4C, none of the pins I tested provided 12V.  I'm not sure what purpose the two additional pins serve.

**Reply by saiarcot895** [reply]
Ooh nice find! I'll try this out sometime next week.

**Reply by bitsmack** [reply]
That is so clean!

I was going to leave extra length of the cable's power wire, like you did, so I didn't need an additional wire running along the outside of the cable. But it hadn't occurred to me to make a disconnect right outside of the RJ-11 jack. :) Thank you.

And thanks to fireheadman for the initial solution and daspiderboris for the GRID header discovery! My M5stack order comes in today :)

**Comment by finishthepint** [comment]
Not 12V but figured I would add how I connected a generic ESP8266 using 5V power built into my water heater.  I have a hybrid heat pump Rheem and found that connector J30, which is accessible without unplugging any wires or removing the back cover, has a 5V pin available.  There is a 5V test point that I used to find this pin.
<img width="1188" height="720" alt="probe" src="https://github.com/user-attachments/assets/780332b1-d10b-4c8d-b3d2-3cf3e4fc795e" />
<img width="496" height="621" alt="wired" src="https://github.com/user-attachments/assets/759833c5-d5f7-4960-8926-740b0bf1a7e2" />

**Comment by k8gg** [comment]
Can confirm what daspiderboris mentioned [above](https://github.com/esphome-econet/esphome-econet/discussions/118#discussioncomment-12370076): the lowest pin of the 4-pin GRID connector does supply 13V DC.

What I did was a dupont female connector to that pin, and I fished the wire through the tiny gap between the (black) RJ12 port and the (grey) housing - I did file the housing to make a bigger, round, smooth hole at the corner so that there is no sharp plastic to pinch any cable anywhere. By the way the pitch of the 4 pins on that GRID / J6 connectors seems to be the same as that of regular dupont connectors.

So no soldering of any kind, and I can roll back to the original condition for any troubleshooting if needed.

== Sidebar question ==
Who knows how to reset the Rheem stats so that I can restart those runtime & kWh numbers, 

== End of sidebar ==

EDIT: Here are my photos hoping to help new users coming here:
![20251227_112730](https://github.com/user-attachments/assets/35ea1a56-a708-4dde-8235-53c2ca0002c8)
![20251227_112919](https://github.com/user-attachments/assets/00eedb7d-a6ff-4bbe-9123-583a8573b445)
![20251227_113208](https://github.com/user-attachments/assets/dc5386d3-6ad8-4187-92ec-6d35609f9e71)
![20251227_114624](https://github.com/user-attachments/assets/dd28c8ac-c75d-4954-9547-a16547b24169)

**Reply by Korkd** [reply]
I went ahead and tested this pin and it showed just under 11v for my unit (XE50T10HS45U1). Should still be fine for supplying power to the A0Z1282CI buck converter on the Atomic RS485 Base, but worth noting that its not the same for every unit.

> Can confirm what daspiderboris mentioned [above](https://github.com/esphome-econet/esphome-econet/discussions/118#discussioncomment-12370076): the lowest pin of the 4-pin GRID connector does supply 13V DC.
> 
> What I did was a dupont female connector to that pin, and I fished the wire through the tiny gap between the (black) RJ12 port and the (grey) housing - I did file the housing to make a bigger, round, smooth hole at the corner so that there is no sharp plastic to pinch any cable anywhere. By the way the pitch of the 4 pins on that GRID / J6 connectors seems to be the same as that of regular dupont connectors.
> 
> So no soldering of any kind, and I can roll back to the original condition for any troubleshooting if needed.
> 
> == Sidebar question == Who knows how to reset the Rheem stats so that I can restart those runtime & kWh numbers,
> 
> == End of sidebar ==
> 
> EDIT: Here are my photos hoping to help new users coming here: ![20251227_112730](https://private-user-images.githubusercontent.com/25398491/530503158-35ea1a56-a708-4dde-8235-53c2ca0002c8.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODg4OTc5MzAsIm5iZiI6MTc4ODg5NzYzMCwicGF0aCI6Ii8yNTM5ODQ5MS81MzA1MDMxNTgtMzVlYTFhNTYtYTcwOC00ZGRlLTgyMzUtNTNjMmNhMDAwMmM4LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA5MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwOTA4VDIwMDAzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI0MDcwOTFiODFlN2RiYjY1ODlhMGJhM2MyYTI2NGJiYzFhODJmNDVjZGZmM2M1NThiZDgxNWI1MDAwMGRhMGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRmpwZWcifQ.XgORO3PiviLOWLWohZdm9A59l_9moRcNYfVGGxO0nHc) ![20251227_112919](https://private-user-images.githubusercontent.com/25398491/530505913-00eedb7d-a6ff-4bbe-9123-583a8573b445.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODg4OTc5MzAsIm5iZiI6MTc4ODg5NzYzMCwicGF0aCI6Ii8yNTM5ODQ5MS81MzA1MDU5MTMtMDBlZWRiN2QtYTZmZi00YmJlLTkxMjMtNTgzYTg1NzNiNDQ1LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA5MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwOTA4VDIwMDAzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0NzQwMWU2OGQ2ZGUyNGMzMjJhYWZlNmIxOWY2MjQwZTE5OGI3ZTAzZDU4NjU5OTFjN2E4NGE4ZTQ3MDk3ODEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRmpwZWcifQ.srHoAs5aD_XMYR-gotbLS8eGDpw2-ckbzLkO-80tBhQ) ![20251227_113208](https://private-user-images.githubusercontent.com/25398491/530505924-dc5386d3-6ad8-4187-92ec-6d35609f9e71.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODg4OTc5MzAsIm5iZiI6MTc4ODg5NzYzMCwicGF0aCI6Ii8yNTM5ODQ5MS81MzA1MDU5MjQtZGM1Mzg2ZDMtNmFkOC00MTg3LTkyZWMtNmQzNTYwOWY5ZTcxLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA5MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwOTA4VDIwMDAzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxMDJkN2Y1MmQ4MjI0MjNhZmY1MGQzYWFiNzYxODhiMzRlOWIzZTBiYzcyMjQxOTljZjAzZWUyZWVmZmIxOTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRmpwZWcifQ.mouZv2l3AYnmpNsZejX-qafpeNXh3SleAQsWOkDmOC0) ![20251227_114624](https://private-user-images.githubusercontent.com/25398491/530505774-dd28c8ac-c75d-4954-9547-a16547b24169.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODg4OTc5MzAsIm5iZiI6MTc4ODg5NzYzMCwicGF0aCI6Ii8yNTM5ODQ5MS81MzA1MDU3NzQtZGQyOGM4YWMtYzc1ZC00OTU0LTk1NDctYTE2NTQ3YjI0MTY5LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA5MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwOTA4VDIwMDAzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUxOWZmZmNhZWIxYjliNzNkNGRmNzQxNjNiODE4MjI3Nzc4YmViYWM0ZTBmNzRlMjAzMmUxNjE3NDY4NGQyNmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRmpwZWcifQ.dG3-qa6ar7byobocoMXNfhqTfnD504XhjgAXUB0SPW0)

**Comment by smnino** [comment]
I have a Rheem Hybrid Heat Pump water heater model PRO H50 T2 RH245 BM. Prior finding this project I was connecting the water heater to the EcoNet cloud via Rheem's wifi module by way of an RJ12 cable. The water heater provided power to the wifi module over the RJ12 cable. 

I used an NodeMCU ESP8266 board with an RS485 and flashed the project's firmware. Once I figure the correct wiring, the new controller integrated into Home Assistant perfectly. My water heater is a 2017 model so, the integration didn't provide all the sensors as I'm sure the newer models do but, it did provide a lot more than the Rheem integration did AND, no more cloud. 

After running the new controller a few days, I got to thinking... I tested and confirmed, pin number 6 out of the water heater's RJ12 connector provides 12v power. I was able to add an MP1584EN step down converter to the case with the ESP8266 to power the ESP8266 and the RS485. 

No need to remove the water heater's control panel and solder to the main board. Power is provided (at least on my model), directly from the RJ12 connection...

**Comment by brandiniman** [comment]
If you don't want to mod and don't have an outlet nearby, I just got a PoE-powered option running great and also used an isolated rs485 board to protect my home network a bit:
https://shop.m5stack.com/products/esp32-ethernet-unit-with-poe
https://shop.m5stack.com/products/isolated-rs485-unit

Yaml so esp32 to rs485 uses provided cable. 
[water-heater1.yaml](https://github.com/user-attachments/files/31658538/water-heater1.yaml)

Cheated and used this https://www.amazon.com/dp/B0F1K78L1S rj12 adapter whose pin map is mislabeled. 
pin1 marked in yellow: 
<img width="549" height="391" alt="image" src="https://github.com/user-attachments/assets/2feba129-cb8d-4c93-91e8-ffb6e80dd436" />

wiring to rs485 board:
<img width="229" height="189" alt="image" src="https://github.com/user-attachments/assets/96de7d5f-b998-453d-ac96-98da5beca34b" />
<img width="342" height="638" alt="image" src="https://github.com/user-attachments/assets/7fb4d3e4-bd7d-438f-9eca-82f63770b3e0" />
